File: backend/src/main/java/stud/ntnu/no/backend/location/service/LocationService.java
package stud.ntnu.no.backend.location.service;

import stud.ntnu.no.backend.location.dto.CreateLocationDTO;
import stud.ntnu.no.backend.location.dto.LocationDTO;

import java.util.List;

/**
 * Service interface for managing locations.
 * <p>
 * This interface defines methods for CRUD operations on locations.
 */
public interface LocationService {
    /**
     * Retrieves all locations.
     *
     * @return a list of LocationDTOs
     */
    List<LocationDTO> getAllLocations();

    /**
     * Retrieves a location by its ID.
     *
     * @param id the ID of the location
     * @return the LocationDTO
     */
    LocationDTO getLocation(Long id);

    /**
     * Creates a new location.
     *
     * @param locationDTO the CreateLocationDTO
     * @return the created LocationDTO
     */
    LocationDTO createLocation(CreateLocationDTO locationDTO);

    /**
     * Updates an existing location.
     *
     * @param id the ID of the location to update
     * @param locationDTO the CreateLocationDTO with updated information
     * @return the updated LocationDTO
     */
    LocationDTO updateLocation(Long id, CreateLocationDTO locationDTO);

    /**
     * Deletes a location by its ID.
     *
     * @param id the ID of the location to delete
     */
    void deleteLocation(Long id);
}

File: frontend/src/api/services/LocationService.ts
import axios from '../axios'

export const LocationService = {
  /**
   * Get all locations
   */
  getAllLocations: () => {
    return axios.get('/api/locations')
  },

  /**
   * Get a location by ID
   */
  getLocationById: (locationId: number) => {
    return axios.get(`/api/locations/${locationId}`)
  },

  /**
   * Create a new location (admin)
   */
  createLocation: (locationData: any) => {
    return axios.post('/api/locations', locationData)
  },

  /**
   * Update a location (admin)
   */
  updateLocation: (locationId: number, locationData: any) => {
    return axios.put(`/api/locations/${locationId}`, locationData)
  },

  /**
   * Delete a location (admin)
   */
  deleteLocation: (locationId: number) => {
    return axios.delete(`/api/locations/${locationId}`)
  },
}

export default LocationService
